File: analytics/src/main/java/com/bluetriangle/analytics/AndroidLogger.java
package com.bluetriangle.analytics;

import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

final public class AndroidLogger extends Logger {
    private static final String tag = "BlueTriangle";
    private final int logLevel;

    public AndroidLogger(int logLevel) {
        this.logLevel = logLevel;
    }

    @Override
    void log(int logLevel, @NonNull String message, @Nullable Object... args) {
        log(logLevel, null, message, args);
    }

    @Override
    void log(int logLevel, @Nullable Throwable throwable, @NonNull String message, @Nullable Object... args) {
        if (logLevel >= this.logLevel) {
            final String formattedMessage = String.format(message, args);
            switch (logLevel) {
                case Log.INFO:
                    Log.i(tag, formattedMessage, throwable);
                    break;
                case Log.WARN:
                    Log.w(tag, formattedMessage, throwable);
                    break;
                case Log.ERROR:
                    Log.e(tag, formattedMessage, throwable);
                    break;
                case Log.DEBUG:
                default:
                    Log.d(tag, formattedMessage, throwable);
                    break;
            }
        }
    }
}

File: analytics/src/main/java/com/bluetriangle/analytics/AndroidLogger.kt
package com.bluetriangle.analytics

import android.util.Log
import com.bluetriangle.analytics.AndroidLogger

internal class AndroidLogger(private val logLevel: Int) : Logger() {
    override fun log(logLevel: Int, message: String) {
        log(logLevel, null, message)
    }

    override fun log(logLevel: Int, throwable: Throwable?, message: String) {
        if (logLevel >= this.logLevel) {
            when (logLevel) {
                Log.INFO -> Log.i(tag, message, throwable)
                Log.WARN -> Log.w(tag, message, throwable)
                Log.ERROR -> Log.e(tag, message, throwable)
                Log.DEBUG -> Log.d(tag, message, throwable)
                else -> Log.d(tag, message, throwable)
            }
        }
    }

    companion object {
        private const val tag = "BlueTriangle"
    }
}